### PR TITLE
Ddp 6323 pex num child answers

### PR DESCRIPTION
**Pull request by evgeniipr**
## Context
 This PR contains implementation of 2 PEX features:
- questions **numChildAnswers** (DDP-6323),
- **user.profile.language** (DDP-6322).

**DDP-6323**
Implement PEX new feature to count number of composite child answers.
Parent question should be of type CompositeQuestion, child answers can be of any types (not only PicklistAnswer) - it's decided to generalize this.
Example:
```
user.studies[“STUDY_GUID”]
    .forms[“ACTIVITY_CODE”]
    .questions[“COMPOSITE_STABLE_ID”]
    .numChildAnswers(“CHILD_STABLE_ID”)
```

**DDP-6322**
Implement PEX new feature to query user's preferred language.
If preferred language is available in user's profile, return it.
Otherwise, throw an exception.
Example:
```
user.profile.language()
```

Updated doc: https://broadinstitute.atlassian.net/wiki/spaces/DDP/pages/1109819399/PEX+Language+Guide

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [x] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [x] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

## FUD Score

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [x] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/pex/TreeWalkInterpreter.java [hunk 1]
```diff
@@ -365,11 +368,12 @@ private Object applyFormInstancePredicate(InterpreterContext ictx, PexParser.For
         }
     }
 
-    private boolean evalQuestionQuery(InterpreterContext ictx, QuestionQueryContext ctx) {
+    private Object evalQuestionQuery(InterpreterContext ictx, QuestionQueryContext ctx) {
         String userGuid = getUserGuidByUserType(ictx, ctx.USER_TYPE());
         String studyGuid = extractString(ctx.study().STR());
         String activityCode = extractString(ctx.form().STR());
         String stableId = extractString(ctx.question().STR());
+        StudyDto studyDto = new JdbiUmbrellaStudyCached(ictx.getHandle()).findByStudyGuid(studyGuid);
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/pex/TreeWalkInterpreter.java]: At this point in the code, we have not checked whether the guids/codes/ids are valid. Someone might accidentally put in a wrong study guid. We should check that the study exists, just like how we're checking the activity exists on L378 below.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/pex/TreeWalkInterpreter.java [hunk 2]
```diff
@@ -379,33 +383,58 @@ private boolean evalQuestionQuery(InterpreterContext ictx, QuestionQueryContext
...
+                        .map(ans -> (CompositeAnswer) ans)
+                        .flatMap(parent -> parent.getValue().stream())
+                        .flatMap(row -> row.getValues().stream())
+                        .filter(child -> child != null && child.getQuestionStableId().equals(childStableId))
+                        .collect(Collectors.toList());
+                childAnswers.forEach(ans -> {
+                    if (ans.getQuestionType() != QuestionType.PICKLIST) {
+                        throw new PexUnsupportedException("For 'numChildAnswers' the children inside composite "
+                                + "should be of type Picklist. Parent stableID: " + stableId);
+                    }
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/pex/TreeWalkInterpreter.java]: I think it's safe to generalize this. We're just counting the number of child answers for a particular child question in the composite. We don't need to restrict it to only picklist child questions.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/pex/TreeWalkInterpreter.java (hunk 1 above)
Comment (evgeniipr) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/pex/TreeWalkInterpreter.java]: Fixed

File: pepper-apis/src/main/java/org/broadinstitute/ddp/pex/TreeWalkInterpreter.java (hunk 2 above)
Comment (evgeniipr) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/pex/TreeWalkInterpreter.java]: Fixed. Updated documentation and PR description.

**Review by yufengwng — APPROVED** [review]

**Pull request merged**